mod_reforged/hooks/config/rumors.nut

```diff
@@ -0,0 +1,68 @@
+            ]);
+            continue;
+        }
+        if (var[0] == "terrain")
+        {
+            local wrongTerrains = [];
+            foreach (terrain in ::Const.Strings.Terrain) if (terrain != var[1] && terrain != "") wrongTerrains.push(terrain);
+            _vars.push([
+                "wrongTerrain",
+                wrongTerrains[::Math.rand(0, wrongTerrains.len() - 1)]
+            ]);
+            continue;
+        }
+    }
+    _vars.push([
+        "legendaryLocationAdjective",
+        ::Const.Strings.LegendaryLocationAdjective[::Math.rand(0, ::Const.Strings.LegendaryLocationAdjective.len() - 1)]
+    ])
+    return buildTextFromTemplate(_text, _vars);
+}
+
+::Const.Strings.LegendaryLocationAdjective <- [
+	"overwhelming",
+	"mystical",
+	"mindblowing",
+	"never before seen",
+	"spectacular",
+    "surreal"
+];
+
+::Const.Strings.RumorsUniqueLocation <- [
+    [   // Doesn't include the name but the information is correct
+        "{I keep hearing talk | Some traveller mentioned something | Some pilgrim mentioned something} about a %legendaryLocationAdjective% {place | location} {%direction% from here | %terrain% | %distance%}"
+    ],
+    [   // Include the exact name but the other information is usually randomly true or fake
+        "{An explorer | A cartographer} just recently came by, said he found {a | some} %legendaryLocationAdjective% {place | location} %distance% {that he | } {called | named} %location%. If only some more details slipped out of his mouth...",
+        "%randomname% told me \'bout %location% the other day. He said it was {%direction% | %wrongDirection% | %wrongDirection%} from here. {But he was drunk so I wouldn\'t trust his word | I\'m probably remembering it wrong}.",
+        "There is an {urban | old} {legend | story} about a {place | location} called %location%. {Some | Some folk} {say | insist} it\'s {%distance% while others estimate it %wrongDistance% | %wrongDistance% while others estimate it %distance%}.",
+        "Just {the other day | yesterday | this morning} a drunk guy {was talking | came up to} to me, said he {found | stumbled upon} {a | some} {place | location} called %location% on one of his adventures. First he claimed it was {%direction%, then he corrected himself saying it was %wrongDirection% | %wrongDirection%, then he corrected himself saying it was %direction%}. In the end he {insisted | sweared} on it being {%direction% | %wrongDirection%}",
+        "{Some | %randomname%s} kid told me {this | a} story {the other day | yesterday | this morning}. It was about {a | some} {place | location} {he calls | called} %location%. He told me it\'s supposed to be {%distance% | %wrongDistance%} to the {%direction% | %wrongDirection%} from here, {%terrain% | %wrongTerrain%}. You {know, I think that story is | believe any of this? Me neither, it\'s} a load of {horse | horse | horse | horse | unhold} shit."
+    ]
+]
```

Comment:
The way I understood you, the message would include both the true and false direction/distance, but for some of these it's randomly true or false.

Reply:
That's correct.

Some messages will contain right&wrong alongside so you will know instantly that one of those must be correct and can go after both hints one after another.
And in other cases you only are given one hint that could be true or false